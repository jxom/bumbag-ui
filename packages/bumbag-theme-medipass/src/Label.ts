import { css } from 'bumbag';

export default {
  styles: {
    base: css`
      font-size: 14px;
    `
  }
};
