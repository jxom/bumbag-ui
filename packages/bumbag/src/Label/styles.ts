import { cssClass } from '../styled';
import { fontSize, theme, fontWeight } from '../utils';

export const Label = (styleProps) => cssClass`
  display: block;
  font-weight: ${fontWeight('semibold')(styleProps)};
  line-height: 1;

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
