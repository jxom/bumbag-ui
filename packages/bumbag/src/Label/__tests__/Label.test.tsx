import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Label } from '../Label';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Label>Hello world</Label>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Label color="primary">Hello world</Label>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a htmlFor attribute', () => {
    const { container } = render(<Label htmlFor="test">Hello world</Label>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Label use="p">Hello world</Label>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Label props', () => {
      const { result } = renderHook(() => Label.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Label>{(LabelProps) => <div {...LabelProps}>Hello world</div>}</Label>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Label.base should render correctly', () => {
    const { container } = render(<Label>hello world</Label>, {
      // @ts-ignore
      theme: { Label: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Label>hello world</Label>, {
      // @ts-ignore
      theme: { Label: { defaultProps: { className: 'test', color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
