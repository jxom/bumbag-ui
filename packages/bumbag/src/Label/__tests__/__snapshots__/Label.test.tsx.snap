// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  padding: 0.05px 0;
}

.emotion-0::before {
  content: '';
  margin-top: -0.2312em;
  display: block;
  height: 0;
}

.emotion-0::after {
  content: '';
  margin-bottom: -0.1921em;
  display: block;
  height: 0;
}

.emotion-0:not(:last-child) {
  margin-bottom: 6.671875px;
}

<p
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Label"
>
  Hello world
</p>
`;

exports[`composition hook should return with Label props 1`] = `
Object {
  "className": "css-17leixx css-63it0p css-jwhr8m bb-Box bb-Label",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  padding: 0.05px 0;
}

.emotion-0::before {
  content: '';
  margin-top: -0.2312em;
  display: block;
  height: 0;
}

.emotion-0::after {
  content: '';
  margin-bottom: -0.1921em;
  display: block;
  height: 0;
}

.emotion-0:not(:last-child) {
  margin-bottom: 6.671875px;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Label"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  padding: 0.05px 0;
}

.emotion-0::before {
  content: '';
  margin-top: -0.2312em;
  display: block;
  height: 0;
}

.emotion-0::after {
  content: '';
  margin-bottom: -0.1921em;
  display: block;
  height: 0;
}

.emotion-0:not(:last-child) {
  margin-bottom: 6.671875px;
}

.emotion-2 {
  color: var(--bb-palette-primary,#000000) !important;
}

<label
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-Label"
>
  hello world
</label>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  padding: 0.05px 0;
}

.emotion-0::before {
  content: '';
  margin-top: -0.2312em;
  display: block;
  height: 0;
}

.emotion-0::after {
  content: '';
  margin-bottom: -0.1921em;
  display: block;
  height: 0;
}

.emotion-0:not(:last-child) {
  margin-bottom: 6.671875px;
}

<label
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Label"
>
  Hello world
</label>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  padding: 0.05px 0;
}

.emotion-0::before {
  content: '';
  margin-top: -0.2312em;
  display: block;
  height: 0;
}

.emotion-0::after {
  content: '';
  margin-bottom: -0.1921em;
  display: block;
  height: 0;
}

.emotion-0:not(:last-child) {
  margin-bottom: 6.671875px;
}

.emotion-2 {
  color: var(--bb-palette-primary,#000000) !important;
}

<label
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Label"
>
  Hello world
</label>
`;

exports[`props should render correctly with a htmlFor attribute 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  padding: 0.05px 0;
}

.emotion-0::before {
  content: '';
  margin-top: -0.2312em;
  display: block;
  height: 0;
}

.emotion-0::after {
  content: '';
  margin-bottom: -0.1921em;
  display: block;
  height: 0;
}

.emotion-0:not(:last-child) {
  margin-bottom: 6.671875px;
}

<label
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Label"
  for="test"
>
  Hello world
</label>
`;

exports[`theming Label.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  padding: 0.05px 0;
}

.emotion-0::before {
  content: '';
  margin-top: -0.2312em;
  display: block;
  height: 0;
}

.emotion-0::after {
  content: '';
  margin-bottom: -0.1921em;
  display: block;
  height: 0;
}

.emotion-0:not(:last-child) {
  margin-bottom: 6.671875px;
}

.emotion-0 {
  background-color: var(--bb-palette-red,#ff0000);
}

<label
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Label"
>
  hello world
</label>
`;
